source/server/listener_manager_impl.cc

```diff
@@ -536,30 +517,13 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
   // Tell all workers to stop accepting new connections on this listener.
   draining_it->listener_->debugLog("draining listener");
   const uint64_t listener_tag = draining_it->listener_->listenerTag();
-  stopListener(
-      *draining_it->listener_,
-      [this,
-       share_socket = draining_it->listener_->listenSocketFactory().sharedSocket().has_value(),
-       listener_tag]() {
-        if (!share_socket) {
```

Comment:
I probable get the answer to my earlier question about dup(). Is it for the purpose that we don't need to check share_socket here? Each worker now owns its own fd to the same accept queue or UDP receive buffer?

Reply:
Yes, exactly. We pre-create all sockets for a couple of reasons:
1) Handle all error handling on the main thread.
2) Unify logic for the most part between reuse port and not reuse port.
3) We need all the sockets on the main thread to handle hot restart for TCP reuse port sockets.